numba/tests/test_exceptions.py

```diff
@@ -430,6 +430,22 @@ def raise_with_no_nrt(i):
         with self.assertRaisesRegex(errors.NumbaRuntimeError, msg):
             raise_with_no_nrt(123)
 
+    def test_try_raise(self):
+
+        @njit
+        def raise_(a):
+            raise ValueError(a)
+
+        @njit
+        def try_raise(a):
+            try:
+                raise_(a)
+            except Exception:
+                pass
+            return a + 1
+
+        self.assertEqual(try_raise.py_func(3), try_raise(3))
```

Comment:
I've got to what I think is the cause of this. Essentially when the code for unwrapping the dynamic exception is generated it has to emit code to box native representations into python objects, but the "environment manager" is missing as kwarg "`env_manager=`" here:
https://github.com/numba/numba/blob/68b0d750eb32e2e1eed50a94062e9bb33c85b52d/numba/core/callconv.py#L582

You can get an environment manager by calling `self.context.get_env_manager(builder)` but this isn't enough though because the emitted environment sentry needs to be aware that this is a Python context code region. I made this change locally to be able to wire this through:
```diff
@@ -123,13 +123,14 @@ class CPUContext(BaseContext):
             builder, envptr, _dynfunc._impl_info['offsetof_env_body'])
         return EnvBody(self, builder, ref=body_ptr, cast_ref=True)
 
-    def get_env_manager(self, builder):
+    def get_env_manager(self, builder, return_pyobject=False):
         envgv = self.declare_env_global(builder.module,
                                         self.get_env_name(self.fndesc))
         envarg = builder.load(envgv)
         pyapi = self.get_python_api(builder)
         pyapi.emit_environment_sentry(
             envarg, debug_msg=self.fndesc.env_name,
+            return_pyobject=return_pyobject
         )
         env_body = self.get_env_body(builder, envarg)
         return pyapi.get_env_manager(self.environment, env_body, envarg)

```
and then did this:
```diff
@@ -576,10 +576,12 @@ class CPUCallConv(BaseCallConv):
         nb_types = [typ for typ in nb_types if typ is not None]
 
         # convert native values into CPython objects
+        env_manager = self.context.get_env_manager(builder,
+                                                   return_pyobject=True)
         objs = []
         for i, typ in enumerate(nb_types):
             val = builder.extract_value(builder.load(st_ptr), i)
-            obj = pyapi.from_native_value(typ, val)
+            obj = pyapi.from_native_value(typ, val, env_manager=env_manager)

```
thoughts?

Reply:
oh, that's fine

Reply:
Thanks @stuartarchibald

Reply:
> Thanks @stuartarchibald

No problem.